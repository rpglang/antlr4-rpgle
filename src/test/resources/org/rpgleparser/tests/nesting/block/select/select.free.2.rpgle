          select;
          when not(*in01) and (fld = 'xx');
          when %sst(a:(x+4):3) = 'ABC';
          when(m) ((Y > 2) and (X<5));
          other;
          endsl;