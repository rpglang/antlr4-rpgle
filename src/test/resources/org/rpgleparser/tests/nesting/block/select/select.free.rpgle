     C                   SELECT
     C                   WHEN      IDCCA1Z = @PRIME1 OR IDCCA1B = @PRIME2 OR
     C                             IDCCA1X = @PRIME3 OR IDCCA1C = @PRIME4
     C                   EVAL      TYPE = '2'

     C                   WHEN      %SUBST(IDHAND:1:1) <> @THANDLER AND
     C                             IDSALE >= @LOW_MI_DSM AND
     C                             %SUBST(IDSALE:1:1) <> %SUBST(@TMREP1:1:1)
     C                   EVAL      TYPE = '3'

     C                   WHEN      %SUBST(IDHAND:1:1) = @THANDLER AND
     C                             IDSALE >= @LOW_MI_DSM AND
     C                             %SUBST(IDSALE:1:1) <> %SUBST(@TMREP1:1:1)
     C                   EVAL      TYPE = '4'

     C                   WHEN      %SUBST(IDHAND:1:1) = @THANDLER AND
     C                             IDSALE >= @TMREP1 AND IDSALE <= @TMREP2
     C                   EVAL      TYPE = '5'
     C                   ENDSL