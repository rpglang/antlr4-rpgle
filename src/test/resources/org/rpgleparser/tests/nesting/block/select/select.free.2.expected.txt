/*===TOKENS===*/
OP_SELECT                <select>
FREE_SEMI                <;>
OP_WHEN                  <when>
NOT                      <not>
'('                      <(>
SPLAT_INDICATOR          <*in01>
')'                      <)>
AND                      <and>
'('                      <(>
ID                       <fld>
'='                      <=>
StringLiteralStart       <'>
StringContent            <xx>
StringLiteralEnd         <'>
')'                      <)>
FREE_SEMI                <;>
OP_WHEN                  <when>
ID                       <%sst>
'('                      <(>
ID                       <a>
COLON                    <:>
'('                      <(>
ID                       <x>
'+'                      <+>
NUMBER                   <4>
')'                      <)>
COLON                    <:>
NUMBER                   <3>
')'                      <)>
'='                      <=>
StringLiteralStart       <'>
StringContent            <ABC>
StringLiteralEnd         <'>
FREE_SEMI                <;>
OP_WHEN                  <when>
'('                      <(>
CS_OperationAndExtender  <m>
')'                      <)>
'('                      <(>
'('                      <(>
ID                       <Y>
'>'                      <>>
NUMBER                   <2>
')'                      <)>
AND                      <and>
'('                      <(>
ID                       <X>
'<'                      <<>
NUMBER                   <5>
')'                      <)>
')'                      <)>
FREE_SEMI                <;>
OP_OTHER                 <other>
FREE_SEMI                <;>
OP_ENDSL                 <endsl>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (statement
    (block
      (selectstatement
        (beginselect (op_select select) ;)
        (whenstatement
          (when
            (op_when
              when
              (indicator_expr
                (expression
                  (expression
                    not
                    (expression ( (expression (identifier (all (symbolicConstants *in01)))) ))
                  )
                  and
                  (expression
                    (
                    (expression
                      (expression (identifier (free_identifier (idOrKeyword fld))))
                      =
                      (expression (literal ' xx '))
                    )
                    )
                  )
                )
              )
            )
            ;
          )
        )
        (whenstatement
          (when
            (op_when
              when
              (indicator_expr
                (expression
                  (expression
                    (function
                      (functionName (free_identifier (idOrKeyword %sst)))
                      (args
                        (
                        (expression (identifier (free_identifier (idOrKeyword a))))
                        :
                        (expression
                          (
                          (expression
                            (expression (identifier (free_identifier (idOrKeyword x))))
                            +
                            (expression (number 4))
                          )
                          )
                        )
                        :
                        (expression (number 3))
                        )
                      )
                    )
                  )
                  =
                  (expression (literal ' ABC '))
                )
              )
            )
            ;
          )
        )
        (whenstatement
          (when
            (op_when
              when
              (cs_operationExtender ( m ))
              (indicator_expr
                (expression
                  (
                  (expression
                    (expression
                      (
                      (expression
                        (expression (identifier (free_identifier (idOrKeyword Y))))
                        (comparisonOperator >)
                        (expression (number 2))
                      )
                      )
                    )
                    and
                    (expression
                      (
                      (expression
                        (expression (identifier (free_identifier (idOrKeyword X))))
                        (comparisonOperator <)
                        (expression (number 5))
                      )
                      )
                    )
                  )
                  )
                )
              )
            )
            ;
            (statement
              (free
                (baseExpression (expression (identifier (free_identifier (op_code other)))))
                ;
              )
            )
          )
        )
        (endselect (op_endsl endsl) ;)
      )
    )
  )
)
/*======*/